packages/flutter_tools/bin/xcode_debug.js

```diff
@@ -0,0 +1,519 @@
+        return command;
+      default:
+        throw `Unrecognized Command: ${command}`;
+    }
+  }
+
+  /**
+   * Validates the flag is allowed for the current command.
+   *
+   * @param {!string} flag
+   * @returns {!bool} The validated command.
+   */
+  isArgumentAllowed(flag) {
+    const allowedArguments = {
+      'common': {
+        '--xcode-path': true,
+        '--project-path': true,
+        '--workspace-path': true,
+        '--verbose': true,
+      },
+      'project-opened': {},
+      'debug': {
+        '--device-id': true,
+        '--scheme': true,
+        '--skip-building': true,
+        '--launch-args': true,
+      },
+      'stop': {
+        '--close-window': true,
+        '--prompt-to-save': true,
+      },
+    }
+
+    if (allowedArguments['common'][flag] === true || allowedArguments[this.command][flag] === true) {
+      return true;
+    }
+    return false;
+  }
+
+  /**
+   * Parses the command line arguments into an object.
+   *
+   * @param {!Array<string>} args List of arguments passed from the command line.
+   * @returns {!Object.<string, string>} Object mapping flag to value.
+   * @throws Will throw an error if flag does not begin with '--'.
+   */
+  parseArguments(args) {
+    const valuesPerFlag = {};
+    for (let index = 1; index < args.length; index++) {
+      const entry = args[index];
+      let flag;
+      let value;
+      const splitIndex = entry.indexOf('=');
+      if (splitIndex === -1) {
+        flag = entry;
+        value = args[index + 1];
+
+        // If next value in the array is also a flag or the next value is
+        // null/undefined, treat the flag like a boolean flag and set the value to 'true'.
+        if ((value != null && value.startsWith('--')) || value == null) {
+          value = 'true';
+        } else {
+          index++;
+        }
+      } else {
+        flag = entry.substring(0, splitIndex);
+        value = entry.substring(splitIndex + 1, entry.length + 1);
+      }
+      if (flag.startsWith('--') === false) {
+        throw `Unrecognized Flag: ${flag}`;
+      }
+
+      valuesPerFlag[flag] = value;
+    }
+    return valuesPerFlag;
+  }
+
+
+  /**
+   * Validates the flag is allowed and `value` is valid. If the flag is not
+   * allowed for the current command, return `null`.
+   *
+   * @param {!string} flag
+   * @param {?string} value
+   * @returns {!string}
+   * @throws Will throw an error if the flag is allowed and `value` is null,
+   *     undefined, or empty.
+   */
+  validatedStringArgument(flag, value) {
+    if (this.isArgumentAllowed(flag) === false) {
+      return null;
+    }
+    if (value == null || value === '') {
+      throw `Missing value for ${flag}`;
+    }
+    return value;
+  }
+
+  /**
+   * Validates the flag is allowed, validates `value` is valid, and converts
+   * `value` to a boolean. A `value` of null, undefined, or empty, it will
+   * return true. If the flag is not allowed for the current command, will
```

Comment:
return false?

Reply:
No, return true. For example, if you `command ---verbose`, `verbose` would be considered true, even though there's no value following it. It's equivalent of `command --verbose=true`